util/git/creds_test.go

```diff
@@ -0,0 +1,122 @@
+package git
+
+import (
+	"github.com/argoproj/argo-cd/v2/util/io"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+)
+
+type cred struct {
+	username string
+	password string
+}
+
+type memoryCredsStore struct {
```

Comment:
does NoopCredsStore test should also be there?

Reply:
Good point. I'll change a couple of the tests to use `NoopCredsStore`. Some of the tests don't check the creds store cleanup, so it doesn't make sense to use the memory creds store.